### PR TITLE
Poset Refactor

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR bundles `Poset` and monotone maps. This is the first stage of the poset refactor: none of the changes around frames/lattices were made yet!.

Todo:
[ ] Update Prose

### CODE REVIEW

File: src/Order/Base.lagda.md [hunk 1]
```diff
@@ -70,158 +74,141 @@ record is-partial-order {ℓ ℓ'} {A : Type ℓ}
...
-  → (f : A → B) → Poset-on ℓ A → Poset-on ℓ' B → Type _
-is-monotone f P Q = ∀ x y → x P.≤ y → f x Q.≤ f y
+private
+  variable
+    o ℓ o' ℓ' o'' ℓ'' : Level
+    P Q R : Poset o ℓ
+
+Monotone-is-hlevel : ∀ n → is-hlevel (Monotone P Q) (2 + n)
+Monotone-is-hlevel {Q = Q} n =
+  Iso→is-hlevel (2 + n) eqv $
```
Comment (plt-amy) [inline src/Order/Base.lagda.md:123]: With the two projections set up I'm pretty sure `hlevel!` could solve this, if you put the lifting to 2+n on the outside

File: src/Order/Base.lagda.md [hunk 2]
```diff
@@ -70,158 +74,141 @@ record is-partial-order {ℓ ℓ'} {A : Type ℓ}
...
+
+instance
+  H-Level-Monotone
+    : ∀ {n}
+    → H-Level (Monotone P Q) (2 + n)
+  H-Level-Monotone = basic-instance 2 (Monotone-is-hlevel 0)
+
+instance
+  Funlike-Monotone : ∀ {o o' ℓ ℓ'} → Funlike (Monotone {o} {o'} {ℓ} {ℓ'})
+  Funlike-Monotone .Funlike.au = Underlying-Poset
```
Comment (plt-amy) [inline src/Order/Base.lagda.md:139]: I'd use a record literal here - Funlike is eta-equality (right?) and the other two fields are instance, so you only have to list `_#_`

File: src/Order/Base.lagda.md [hunk 3]
```diff
@@ -70,158 +74,141 @@ record is-partial-order {ℓ ℓ'} {A : Type ℓ}
...
+Monotone-pathp {P = P} {Q} {f} {g} q i .Monotone.pres-≤ {x} {y} α =
+  is-prop→pathp
+    (λ i → Π-is-hlevel³ {A = ⌞ P i ⌟} {B = λ _ → ⌞ P i ⌟} {C = λ x y → P i .Poset._≤_ x y} 1
+      λ x y _ → Q i .Poset.≤-thin {q i x} {q i y})
+    (λ _ _ α → f .Monotone.pres-≤ α)
+    (λ _ _ α → g .Monotone.pres-≤ α) i x y α      
+
+Extensional-Monotone
+  : ∀ {o ℓ o' ℓ' ℓr} {P : Poset o ℓ} {Q : Poset o' ℓ'}
+  → ⦃ sq : Extensional (⌞ Q ⌟) ℓr ⦄
```
Comment (plt-amy) [inline src/Order/Base.lagda.md:158]: If you only ask for extensionality on the codomain then you lose the possibility of having specialisations for `P → Q`, e.g. if `P` is a quotient

File: src/Order/Instances/Pointwise.lagda.md [hunk 4]
```diff
@@ -48,11 +49,11 @@ another poset, you might want to consider the full subposet of $P \to Q$
 consisting of the monotone maps:
 
 ```agda
-Monotone : ∀ {ℓₒ ℓᵣ ℓₒ' ℓᵣ'}
```
Comment (plt-amy) [inline src/Order/Instances/Pointwise.lagda.md:51]: thoughts on calling this `Poset[_,_]` like we have `Cat[_,_]`?

File: src/Order/Base.lagda.md [hunk 5]
```diff
@@ -151,17 +142,17 @@ Monotone-pathp {P = P} {Q} {f} {g} q i .Monotone.pres-≤ {x} {y} α =
...
 
 Extensional-Monotone
   : ∀ {o ℓ o' ℓ' ℓr} {P : Poset o ℓ} {Q : Poset o' ℓ'}
-  → ⦃ sq : Extensional (⌞ Q ⌟) ℓr ⦄
-  → Extensional (Monotone P Q) (o ⊔ ℓr)
+  → ⦃ sa : Extensional (⌞ P ⌟ → ⌞ Q ⌟) ℓr ⦄
+  → Extensional (Monotone P Q) ℓr
 Extensional-Monotone {Q = Q} ⦃ sq ⦄ =
-  injection→extensional! Monotone-pathp (Extensional-→ ⦃ sq ⦄)
+  injection→extensional! Monotone-pathp sq
```
Comment (favonia) [inline src/Order/Base.lagda.md:152]: @plt-amy coding style question: should `sa` and `sq` be unified? Actually, what does `s` stand for? (I had this question when playing with automation in 1lab.)

Comment (plt-amy) [reply src/Order/Base.lagda.md:152]: > should `sa` and `sq` be unified?

Probably - there's nothing in the style guide about this, but maybe there should be?

> Actually, what does `s` stand for?

Originally the `Extensional` class was called `Setlike`, so the s is for set(oid). But now it doesn't exactly stand for anything :slightly_smiling_face:

File: src/Order/Morphism.lagda.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    → ∀ {x y} → (f # x) Q.≤ (f # y) → x P.≤ y
+  section→reflect-≤ f g sect {x = x} {y = y} fx≤fy =
+    x                        P.=˘⟨ sect x ⟩
+    g # (f # x)              P.≤⟨ g .pres-≤ fx≤fy ⟩
+    g # (f # y)              P.=⟨ sect y ⟩
+    y                        P.≤∎
+```
+
+As a corollary, if $f : P \to Q$ is a section, then $f$ reflects the
+the orderings on $P$ is equivalent with its image in $Q$.
```
Comment (ncfavier) [inline src/Order/Morphism.lagda.md:53]: f reflects the the orderings?

Comment (plt-amy) [reply src/Order/Morphism.lagda.md:53]: This entire sentence is a trainwreck

File: src/Order/Univalent.lagda.md
Comment (ncfavier) [inline src/Order/Univalent.lagda.md:1]: Should this be `Order.Cat.Univalent`? The file name made me think it was about univalent preorders.

File: src/Order/Base.lagda.md [hunk 7]
```diff
@@ -11,55 +14,120 @@ module Order.Base where
...
+notions of "order" in mathematics:
+
+- The "usual" ordering relations $x \le y$ on familiar number systems
+  like [the natural numbers], [the integers], the rationals, or the
+  reals.
+
+  When ordering the naturals, the integers, and the rationals, we can
+  say more: they are _linear_ orders. That is, in addition to the
+  properties of $\le$ required to be a poset, we have, for any pair of
+  elements,
```
Comment (favonia) [inline src/Order/Base.lagda.md:32]:
```suggestion
  elements $x$ and $y$,
```

File: src/Order/Base.lagda.md [hunk 8]
```diff
@@ -11,55 +14,120 @@ module Order.Base where
...
+which, _a priori_, might not be a proposition: we have not included the
+assumption that `Ob`{.Agda} is actually a set. Therefore, it might be
+the case that an identification between posets does _not_ correspond to
+an identification of the underlying types and one of the relation.
+However, since the "symmetric part" of $\le$, the relation $x \sim y$
+iff.
+
+$$
+(x \le y) \land (y \le x)\text{,}
+$$
```
[the remark this answers is not in the document]
Comment (favonia) [reply src/Order/Base.lagda.md:122]: Actually, let me try again...

**Comment by favonia** [comment]
Question: why do we want `opaque` when, as far as I can tell, no one will want to unfold `Ob-is-set`?

**Review by plt-amy — APPROVED** [review]
